parsing/pprintast.ml

```diff
@@ -1255,7 +1255,16 @@ and binding ctxt f {pvb_pat=p; pvb_expr=x; _} =
       Some (p, pt_tyvars, e_ct, e) else None
     | _ -> None in
   if x.pexp_attributes <> []
-  then pp f "%a@;=@;%a" (pattern ctxt) p (expression ctxt) x else
+  then
+    match p with
+    | {ppat_desc=Ppat_constraint({ppat_desc=Ppat_var _; _} as pat,
+                                 ({ptyp_desc=Ptyp_poly _; _} as typ));
+       ppat_attributes=[]; _} ->
```

Comment:
I don't understand why we match on `ppat_attributes=[]` here.

Reply:
There's no good way to print attributes here (they won't get re-parsed to be against this node), so we fall back to the invalid pattern printing which accurately represents the state of the AST.

This should only really arise if a ppx author has contrived it, in which case being able to see the attribute probably has more value than parsability after printing.

Reply:
Thanks, makes sense.